core-backend/sandbox/src/funcs.rs

```diff
@@ -626,42 +603,33 @@ impl<E: Ext + 'static> FuncsHandler<E> {
 
         let value_ptr = pop_i32(&mut args)?;
 
-        ctx.ext
-            .clone()
-            .with(|ext| funcs::set_u128(&mut ctx.memory, value_ptr, ext.value_available()))
-            .and_then(|res| {
-                res.map(|_| ReturnValue::Unit)
-                    .map_err(|_| "Cannot set u128")
-            })
+        let Runtime { ext, memory, .. } = ctx;
+
+        ext.with_fallible(|ext| funcs::set_u128(memory, value_ptr, ext.value_available()))
+            .map(|_| ReturnValue::Unit)
             .map_err(|err| {
                 ctx.trap = Some(err);
                 HostError
             })
     }
 
     pub fn leave(ctx: &mut Runtime<E>, _args: &[Value]) -> SyscallOutput {
-        let _: Result<ReturnValue, HostError> = ctx
-            .ext
-            .with(|ext| ext.leave())
-            .map(|_| return_none())
-            .map_err(|err| {
-                ctx.trap = Some(err);
-                HostError
-            })?;
-        ctx.trap = Some(LEAVE_TRAP_STR);
+        if let Err(err) = ctx.ext.with_fallible(|ext| ext.leave()) {
+            ctx.trap = Some(err);
+        } else {
+            ctx.trap = Some(LEAVE_TRAP_STR);
+        }
```

Comment:
Not shorter but maybe a little bit more readable.

```suggestion
        ctx.trap = ctx
            .ext
            .with_fallible(|ext| ext.leave())
            .err()
            .or(Some(LEAVE_TRAP_STR));
```

Reply:
done in [0817d34](https://github.com/gear-tech/gear/pull/853/commits/0817d34d8fe8168d3d646a8c27dd205e507d585b)